### PR TITLE
Add `slugify` Jinja filter

### LINKED ISSUE #4477: Expose django slugify in jinja filters
As a Network Automation Engineer, I want the django slugify utility function to be exposed for use in Nautobot jinija2 templates, so that I can have feature parity with the fact it is already exposed as a django template filter.

I will know this is complete when the django slugify function is exposed a registered jinja2 filter.

### PR DESCRIPTION
**Pull request by bryanculver**
# Closes: #4477
# What's Changed

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/dcim/templates/dcim/device.html [hunk 1]
```diff
@@ -14,7 +14,7 @@
                         <div class="col-md-6">
                             <div class="panel panel-default">
                                 <div class="panel-heading">
-                                    <strong>Device</strong>
+                                    <strong>Device {{ "ams 01" | slugify }}</strong>
```
Comment (glennmatthews) [inline nautobot/dcim/templates/dcim/device.html:17]: :-)

File: changes/4477.added [hunk 2]
```diff
@@ -0,0 +1 @@
+Added the `slugify_dashes_to_underscores` helper function as a Jinja filter called `slugify`.
```
Comment (glennmatthews) [inline changes/4477.added:1]:
```suggestion
Added the Django `slugify` helper function as a Jinja filter.
```

**Review by glennmatthews — APPROVED** [review]

File: changes/4477.added (hunk 2 above)
Comment (bryanculver) [reply changes/4477.added]: image
Acceptable?

File: nautobot/dcim/templates/dcim/device.html (hunk 1 above)
Comment (bryanculver) [reply nautobot/dcim/templates/dcim/device.html]: 🤦

**Pull request merged**